### PR TITLE
Enable PROCGETSTORAGE on CACHE_HIT

### LINKED ISSUE #2806: Cache problems
I still highly suspect that these disabled parts of `kdbGet` will cause problems with `procgetstroage` plugins.

https://github.com/ElektraInitiative/libelektra/blob/2734034e287fd6d657ce08b5257a6e7836569cf1/src/libs/elektra/kdb.c#L1005-L1016

However, I cannot verify that, because I simply cannot produce a cache hit. This is because a cache miss is caused, if the resolver returns NO_UPDATE for any of the backends.

https://github.com/ElektraInitiative/libelektra/blob/2734034e287fd6d657ce08b5257a6e7836569cf1/src/libs/elektra/kdb.c#L498-L500

A cache hit only happens, if CACHE_HIT was returned for all backends.

https://github.com/ElektraInitiative/libelektra/blob/2734034e287fd6d657ce08b5257a6e7836569cf1/src/libs/elektra/kdb.c#L510-L514

Not sure, whether this is intended or not. I also couldn't find any test cases for the cache, through which I could find out how to get a cache hit.

---

My suspicion of the bug in `kdbGet` should be verifiable with the `examples/gopts` example. If there is some way that its `kdbGet` call executes the "cache hit" or "no update" branches, the `gopts` plugin won't be called and command-line options will not be reported correctly.

### PR DESCRIPTION
**Pull request by mpranj**
Closes #2806.

## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [x] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [x] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests
- [x] I ran all tests locally and everything went fine
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] meta data is updated (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))

## Review

Reviewers will usually check the following:

- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)
- [ ] Documentation is concise and good to read
- [ ] Examples are well chosen and understandable

## Labels

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the build server is happy** and also you
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by mpranj** [comment]
Done. I hope this implements what @kodebach needed.

File: src/libs/elektra/kdb.c [hunk 1]
```diff
@@ -810,11 +817,16 @@ static int elektraCacheLoadSplit (KDB * handle, Split * split, KeySet * ks, KeyS
 	if (debugGlobalPositions)
 	{
 		keySetName (parentKey, keyName (initialParent));
-		elektraGlobalGet (handle, *cache, parentKey, PROCGETSTORAGE, INIT);
-		elektraGlobalGet (handle, *cache, parentKey, PROCGETSTORAGE, MAXONCE);
-		elektraGlobalGet (handle, *cache, parentKey, PROCGETSTORAGE, DEINIT);
+		elektraGlobalGet (handle, ks, parentKey, PREGETSTORAGE, INIT);
+		elektraGlobalGet (handle, ks, parentKey, PREGETSTORAGE, MAXONCE);
+		elektraGlobalGet (handle, ks, parentKey, PREGETSTORAGE, DEINIT);
```
Comment (kodebach) [inline src/libs/elektra/kdb.c]: I haven't tested this PR yet, but it seems weird to me that you use `ks` here and `*cache` in line 825.

File: src/libs/elektra/kdb.c [hunk 2]
```diff
@@ -1130,6 +1142,7 @@ int kdbGet (KDB * handle, KeySet * ks, Key * parentKey)
 	if (cacheData && handle->globalPlugins[POSTGETCACHE][MAXONCE])
 	{
 		splitCacheStoreState (handle, split, handle->global, cacheParent, initialParent);
+		elektraCutProc (ks); // remove proc keys before caching
```
Comment (kodebach) [inline src/libs/elektra/kdb.c]: Again, haven't tested, but doesn't this line mean that the `proc` keys won't be returned by `kdbGet`. As far as I can tell from the code they are removed, but not added again.

Comment (mpranj) [reply src/libs/elektra/kdb.c]: You're right. I thought the proc keys are only internal. I'll fix it.

**Comment by mpranj** [comment]
@kodebach thank you for the review. It should be fixed now.

**Comment by markus2330** [comment]
Thank you!

@kodebach could you please also test and (re)open the issue if it persists?

**Comment by kodebach** [comment]
It works now.

**Pull request merged**